frontend/src/components/pages/Scheduling/SelectDateTime.tsx

```diff
@@ -91,24 +95,29 @@ const SelectDateTime = ({
   const [showTimeSlots, setShowTimeSlots] = useState<string[] | null>(
     getTimeSlot(dayPart),
   );
+  const [frequencyLabels, setFrequencyLabels] = useState<string[]>(frequencies);
+  const [showFrequency, setShowFrequency] = useState<boolean>(
+    false
+  );
+  const [showTimeofDay, setShowTimeofDay] = useState<boolean>(
+    false
+  );
+  const [canSubmit, setCanSubmit] = useState<boolean>(false);
+
   const [icons, setIcons] = useState<number[]>([0, 0, 0, 0, 0]);
   const [schedules, setSchedules] = useState<Schedule[]>([]);
   const [isOneTimeDonation, setIsOneTimeDonation] = useState<boolean>(
-    frequency === "One time" || frequency === "",
-  );
-  const [recurringEndDate, setRecurringEndDate] = useState<string>(
-    recurringDonationEndDate
-      ? moment(recurringDonationEndDate).format("MM/DD/YYYY")
-      : "",
+    frequency === frequencies[0] || frequency === "",
   );
+  const [recurringEndDate, setRecurringEndDate] = useState<Date>(new Date(startTime));
 
   React.useEffect(() => {
     // Need to set frequency here since "One time"
     // is default value in desktop dropdown
     if (isDesktop && isOneTimeDonation) {
-      setForm({ target: { name: "frequency", value: "One time" } });
+      setForm({ target: { name: "frequency", value: frequencies[0] } });
```

Comment:
this is probably more of a design note but i found it a little weird that all the other options were not automatically filled out but this one is - curious if there was a reason for this?

Reply:
i think this is part of the reason why the bug is introduced

Reply:
i believe this was implemented because the "Frequency" input for desktop used to be a dropdown. But now they are radio buttons, so it should be fine if I remove it. Thanks for pointing that out!